[Android] expo prebuilt should include react.codegenDir in app/build.gradle to support monorepos

### Summary

When using `expo prebuild` the `app/build.gradle`'s react section looks like this:

```groovy
react {
    entryFile = file(["node", "-e", "require('expo/scripts/resolveAppEntry')", projectRoot, "android", "absolute"].execute(null, rootDir).text.trim())
    reactNativeDir = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
    hermesCommand = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsolutePath() + "/sdks/hermesc/%OS-BIN%/hermesc"
    debuggableVariants = expoDebuggableVariants

    /* Folders */
    //   The root of your project, i.e. where "package.json" lives. Default is '..'
    // root = file("../")
    //   The folder where the react-native NPM package is. Default is ../node_modules/react-native
    // reactNativeDir = file("../node_modules/react-native")
    // ⚠️👀 Read the following line carefully:
    //   The folder where the react-native Codegen package is. Default is ../node_modules/react-native-codegen
    // codegenDir = file("../node_modules/react-native-codegen")
```

We "dynamically" resolve the path to the react native directory. This is good and important for project setups where the `react-native` package isn't inside the app's `node_modules/` folder.

However, we need the same dynamic resolving for the codegen path.

I was recently working in a mono repo where the build failed with the following error:

```
> Task :react-native-blob-jsi-helper:buildCodegenCLI FAILED
FAILURE: Build failed with an exception.
* What went wrong:
Execution failed for task :react-native-blob-jsi-helper:buildCodegenCLI 
> A problem occurred when starting the process 'command' '/apps/expo/node modules/react-native-codegen/scripts/oss/build.sh''
```

> A little insight in this issue, as /apps/expo/node modules/react-native-codegen/lib doesn't exist the react-native-gradle-plugin wants to build the codegen cli from source, which fails. This is all due to the fact that it looks for `react-native-codegen` in the wrong place.

This was due to the fact that it looks in the wrong `node_modules` folder for the react-native-codegen package.
We can make sure we stay compatible with monorepo structures by simply adding the following line:

```groovy
react {
    entryFile = file(["node", "-e", "require('expo/scripts/resolveAppEntry')", projectRoot, "android", "absolute"].execute(null, rootDir).text.trim())
    reactNativeDir = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
    codegenDir = new File(["node", "--print", "require.resolve('react-native-codegen/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
```

I am not sure where the template files for `expo prebuild` are located, otherwise I'd have opened a PR myself :swe

### What platform(s) does this occur on?

Android

### SDK Version

48

### Environment

  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 13.2
      Shell: 5.8.1 - /bin/zsh
    Binaries:
      Node: 16.15.1 - ~/.nvm/versions/node/v16.15.1/bin/node
      Yarn: 3.5.0 - ~/.nvm/versions/node/v16.15.1/bin/yarn
      npm: 8.11.0 - ~/.nvm/versions/node/v16.15.1/bin/npm
    Managers:
      CocoaPods: 1.12.0 - /Users/hannomargelo/.rbenv/shims/pod
    SDKs:
      iOS SDK:
        Platforms: DriverKit 22.2, iOS 16.2, macOS 13.1, tvOS 16.1, watchOS 9.1
      Android SDK:
        Android NDK: 21.4.7075529
    IDEs:
      Android Studio: Electric Eel 2022.1.1 Patch 2 Electric Eel 2022.1.1 Patch 2
      Xcode: 14.2/14C18 - /usr/bin/xcodebuild
    npmGlobalPackages:
      eas-cli: 3.9.0
    Expo Workflow: bare


### Minimal reproducible example

I don't think a reproduction is needed as from the above examples it can be seen why adding the codegen path can become necessary. This is also more of a suggestion than a real bug.

--- a/templates/expo-template-bare-minimum/android/app/build.gradle
+++ b/templates/expo-template-bare-minimum/android/app/build.gradle
@@ -21,6 +21,7 @@ react {
     entryFile = file(["node", "-e", "require('expo/scripts/resolveAppEntry')", projectRoot, "android", "absolute"].execute(null, rootDir).text.trim())
     reactNativeDir = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
     hermesCommand = new File(["node", "--print", "require.resolve('react-native/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsolutePath() + "/sdks/hermesc/%OS-BIN%/hermesc"
+    codegenDir = new File(["node", "--print", "require.resolve('react-native-codegen/package.json')"].execute(null, rootDir).text.trim()).getParentFile().getAbsoluteFile()
     debuggableVariants = expoDebuggableVariants
 
     // Use Expo CLI to bundle the app, this ensures the Metro config

